src/app/clusters/diagnostic-logs-server/BDXDiagnosticLogsProvider.h

```diff
@@ -58,6 +58,15 @@ class BDXDiagnosticLogsProvider : public bdx::Initiator
      */
     void HandleTransferSessionOutput(bdx::TransferSession::OutputEvent & event) override;
 
+    /**
+     * This method returns a singleton instance of this class.
+     */
+    static BDXDiagnosticLogsProvider & GetInstance()
```

Comment:
This should not be a singleton. If anything it should be injected in Server.cpp. The fact that it was previously a global variable is not good, but it's something that should be fixed without having a new method to more easily  access the BDXDiagnosticsLogsProivider gllobally.